lib/stat_tracker.rb

```diff
@@ -23,6 +23,40 @@ def initialize(files)
     end
   end
 
+  def most_accurate_team(season)
+  # name of team with the best ratio of shots to goals for the season 
+  # 1. pull out all game_id for a season in @games as an array
+  # 2. iterate over @game_teams using our created array of game_id as reference
+    # make hash of team_id as keys and set values to [goals + itself, shots + itself]
+  # 3. use transform_values on hash to give goals / shots .round(4)
+  # 4. Find the max value and subsequent key and return the team_name with team_list
+
+  # 1. array of all game_id from a season
+  all_season_game_id = @games.map do |game|
+    game.game_id if game.season == season
+  end.compact
```

Comment:
Did you have to use compact because some values returned nil? What values were those?

Reply:
Compact is needed to remove all nil values that result from not passing the conditional logic on line 36.